sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -672,6 +667,15 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
     }
   }
 
+  private def fixUpBroadcastHashJoin(): Unit = {
+    childPlans.foreach(_.fixUpBroadcastHashJoin())
+    if(wrapped.isInstanceOf[BroadcastHashJoinExec]) {
+      // Check the tagging if it can not run on GPU, because this fallback may
+      // be caused by InputFileBlockRule.
+      this.asInstanceOf[GpuBroadcastHashJoinMetaBase].checkTagForBuildSide()
```

Comment:
Is this only impact ```BroadcastHashJoinExec```? How about ```BoardcastNestedLoopJoin```? It seems to have similar problem there?

Reply:
yeah, i just found this. will update it